proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,793 @@
+                                                 Moderate   https://github.com/advisories/GHSA-v76m-f5cx-8rg4
+                                                 Critical   https://github.com/advisories/GHSA-x8f7-h444-97w4
+                                                 Moderate   https://github.com/advisories/GHSA-5c66-x4wm-rjfx
+                                                 High       https://github.com/advisories/GHSA-x2rg-fmcv-crq5
+                                                 High       https://github.com/advisories/GHSA-j3g9-6fx5-gjv7
+                                                 High       https://github.com/advisories/GHSA-xx3h-j3cx-8qfj
+                                                 Moderate   https://github.com/advisories/GHSA-5whq-j5qg-wjvp
+   > DotNetZip            1.0.0       1.0.0      High       https://github.com/advisories/GHSA-7378-6268-4278
+
+Project `MyProjectB` has the following vulnerable packages
+   [netcoreapp3.1]:
+   Top-level Package      Requested   Resolved   Severity   Advisory URL
+   > DotNetNuke.Core      6.0.0       6.0.0      High       https://github.com/advisories/GHSA-g8j6-m4p7-5rfq
+                                                 Moderate   https://github.com/advisories/GHSA-v76m-f5cx-8rg4
+                                                 Critical   https://github.com/advisories/GHSA-x8f7-h444-97w4
+                                                 Moderate   https://github.com/advisories/GHSA-5c66-x4wm-rjfx
+                                                 High       https://github.com/advisories/GHSA-x2rg-fmcv-crq5
+                                                 High       https://github.com/advisories/GHSA-j3g9-6fx5-gjv7
+                                                 High       https://github.com/advisories/GHSA-xx3h-j3cx-8qfj
+                                                 Moderate   https://github.com/advisories/GHSA-5whq-j5qg-wjvp
+
+   [net5.0]:
+   Top-level Package      Requested   Resolved   Severity   Advisory URL
+   > DotNetNuke.Core      6.0.0       6.0.0      High       https://github.com/advisories/GHSA-g8j6-m4p7-5rfq
+                                                 Moderate   https://github.com/advisories/GHSA-v76m-f5cx-8rg4
+                                                 Critical   https://github.com/advisories/GHSA-x8f7-h444-97w4
+                                                 Moderate   https://github.com/advisories/GHSA-5c66-x4wm-rjfx
+                                                 High       https://github.com/advisories/GHSA-x2rg-fmcv-crq5
+                                                 High       https://github.com/advisories/GHSA-j3g9-6fx5-gjv7
+                                                 High       https://github.com/advisories/GHSA-xx3h-j3cx-8qfj
+                                                 Moderate   https://github.com/advisories/GHSA-5whq-j5qg-wjvp
+```
+
+### dotnet list package --vulnerable --json
+
+```json
```

Comment:
How can we record in the output that `--include-transitive` wasn't used here?
In other words, if I look at this output years from now, how would I know whether any transitives were in this project?

Not sure how best to indicate that in the JSON itself. 
Perhaps the missing "transitivePackages" section as it is explains this? Is that intuitive, consistent with other JSON-output CLIs?
Certainly, output for 0 transitives with `--include-transitives` should not look identical when omitting `--include-transitives`.

Reply:
> I don't think it's **ideal** but --include-transitive I would naively expect to have an empty array when there are 0 transitive dependencies vs not having a key at all. Though I would probably argue that --include-transitive is a weird flag to have. I think it makes more sense to include them by default (since it's already separated into a different group). I think if I'm looking for vulns it would be a real footgun to have to say "but really all the vulns please" at the end of my command

I agree. I would expect transitive dependencies to be included by default.

Reply:
Yes, I start to tilt toward including transitive dependencies by default because output should be full snapshot of SBOM, without transitive dependencies it's not full, because dependencies could have vulns.

Reply:
After offline discussion I strongly agree with `The behavior of the command is separate from the json output.` 
I created separate tracking issue: https://github.com/NuGet/Home/issues/11550

Reply:
See other discussion.